### PR TITLE
Reset the repository

### PR DESCRIPTION
**Pull request by kylelaker-ccpo**

### CODE REVIEW

File: test/atat-web-api.test.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// import * as cdk from 'aws-cdk-lib';
+// import { Template } from 'aws-cdk-lib/assertions';
+// import * as AtatWebApi from '../lib/atat-web-api-stack';
+
+// example test. To run these tests, uncomment this file along with the
+// example resource in lib/atat-web-api-stack.ts
+test('SQS Queue Created', () => {
```
Comment (jeffsegal-ccpo) [inline test/atat-web-api.test.ts:7]: Is this boilerplate or are we trying to actually have proper unit tests to validate our templates? Would be pretty cool if it's useful and not a huge burden.

**Review by jeffsegal-ccpo — CHANGES_REQUESTED** [review]
Looks like a nice clean start!

File: test/atat-web-api.test.ts (hunk 1 above)
Comment (kylelaker-ccpo) [reply test/atat-web-api.test.ts:7]: I really want to have real tests! We don't have any infrastructure defined in this PR and this _particular_ code is boilerplate (from a `cdk init`) but the `aws-cdk-lib/assertions` library is pretty awesome!

Comment (kylelaker-ccpo) [reply test/atat-web-api.test.ts:7]: Ya know what, let's get off to to a good start on this reset. I will define some basic infra to kick off at least a few tests

File: lib/constructs/apigateway.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      },
+      policy: this.privateEndpointPolicy(props.vpcConfig?.interfaceEndpoint!),
+    };
+    const regionalEndpointConfig = {
+      endpointTypes: [apigw.EndpointType.REGIONAL]
+    }
+
+    const restApi = new apigw.RestApi(this, "Api", {
+      ...props,
+      ...(isPrivateApi ? privateEndpointConfig : regionalEndpointConfig),
```
[the remark this answers is not in the document]
Comment (kylelaker-ccpo) [reply lib/constructs/apigateway.ts:67]: Turns out it doesn't work. Good thing I'm working on tests!

**Review by jeffsegal-ccpo — APPROVED** [review]

**Review by juliusfitzhugh-ccpo — APPROVED** [review]
Nice wipe of the repo! I assume the repo looked similar to this when the team switched over to AWS. I like the inclusion of more tests for our infrastructure. Ran `npm run test` locally and all tests passed. Also did a quick `cdk diff` which produced mostly expected results. Good base to build upon.

File: lib/constructs/apigateway.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (kylelaker-ccpo) [reply lib/constructs/apigateway.ts]: And for clarity, the general structure worked but the configuration creation for each path needs to be a function invoked conditionally so we don't crash on a type error when things are undefined.

File: lib/atat-net-stack.ts [hunk 3]
```diff
@@ -1,19 +1,39 @@
...
+
+const DEFAULT_ROUTE = "0.0.0.0/0";
 
 export interface AtatNetStackProps extends cdk.StackProps {
+  /**
+   * The CIDR block to use for the VPC.
+   *
+   * If this is not provided, the default value from the CDK's VPC construct
+   * will be used. VPCs with overlapping ranges may cause routing issues for
+   * the application. This value should almost always
```
Comment (jasonburkert-ccpo) [inline lib/atat-net-stack.ts:15]: Unfinished sentence here.  _Should almost always..._

**Review by jasonburkert-ccpo — APPROVED** [review]
Nice.  Left one minor comment.

**Review by zachclark-ccpo — APPROVED** [review]

File: lib/atat-net-stack.ts (hunk 3 above)
Comment (kylelaker-ccpo) [reply lib/atat-net-stack.ts]: Good catch. Changed to:

>    * the application. This value should almost always be provided.

**Pull request merged**